src/block_iterator.rs

```diff
@@ -47,6 +48,7 @@ pub struct BlockIterator<B: BlockLike> {
     // first key in the block, because slateDB does not support multi version of keys
     // so we use `Bytes` temporarily
     first_key: Bytes,
+    ordering: IterationOrder,
```

Comment:
curious whether it's intentional that `BlockIterator` is `pub` but `IterationOrder` is `pub(crate)`? Is `BlockIterator` part of the public API?

Reply:
Yeah, I noticed that too. Seems like it should be crate public, but will let @rodesai or @criccomini chime in.

Reply:
Yea, I think it should be `pub(crate)`. Noticing a few things here and there where we've gone with public instead of `pub(crate)` by accident.